engine/src/main/java/org/terasology/engine/rendering/opengl/GLSLShader.java

```diff
@@ -278,7 +278,8 @@ private void registerAllShaderPermutations() {
             disposalAction.vertexPrograms.put(featureHash, vertShaderId);
         }
 
-        logger.debug("Compiled {} permutations for {}.", allPermutations.size(), getUrn());
+        logger.atDebug().addArgument(() -> allPermutations.size()).addArgument(() -> getUrn()).
+                log("Compiled {} permutations for {}.");
```

Comment:
`getUrn()` is a simple getter method and `size()` complexity is probably O(1) I'd say so impact of both are neglectable
```suggestion
        logger.debug("Compiled {} permutations for {}.", allPermutations.size(), getUrn()); //NOPMD
```